server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -6,86 +6,82 @@
 from typing import List
 
 from pbench.process_tb import ProcessTb, Results
-from pbench.test import tarball, bad_tarball, fixture_path
+from pbench.test import tarball, bad_tarball, receive_dir
 
 
 class MockConfig:
     TS = "run-1970-01-02T00:00:00.000000"
 
+    def __init__(self, mappings: dict):
+        self.mappings = mappings
+
     def get(self, section: str, option: str) -> str:
-        return ""
+        assert (
+            section in self.mappings and option in self.mappings[section]
+        ), f"Unexpected configuration option, {section!r}/{option!r}, in MockConfig.get()"
+        return self.mappings[section][option]
 
 
 class TestProcessTb:
     def test_token(self, monkeypatch, test_logger):
-        def mock_get(self, section: str, option: str) -> str:
-            assert option in [
-                "pbench-receive-dir-prefix",
-                "put-token",
-            ], f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
-            return fixture_path if option == "pbench-receive-dir-prefix" else ""
+        def mock_is_dir(self) -> bool:
+            return True
 
         expected_error_msg = (
             "No value for config option put-token in section pbench-server"
         )
-        monkeypatch.setattr(MockConfig, "get", mock_get)
+        mappings = {
+            "pbench-server": {
+                "pbench-receive-dir-prefix": receive_dir,
+                "put-token": "",
+            }
+        }
+        monkeypatch.setattr(Path, "is_dir", mock_is_dir)
         with pytest.raises(ValueError) as e:
-            ProcessTb(MockConfig(), test_logger)
+            ProcessTb(MockConfig(mappings), test_logger)
 
         assert expected_error_msg in str(e)
 
-    def test_get_receive_dir_failed(self, monkeypatch, test_logger):
-        def mock_get(self, section: str, option: str) -> str:
-            assert (
-                option == "pbench-receive-dir-prefix"
-            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
-            return ""
-
+    def test_get_receive_dir_failed(self, test_logger):
         expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
-        monkeypatch.setattr(MockConfig, "get", mock_get)
+        mappings = {
+            "pbench-server": {
+                "pbench-receive-dir-prefix": "",
+                "put-token": "Authorization-token",
+            }
+        }
         with pytest.raises(ValueError) as e:
-            ProcessTb(MockConfig(), test_logger)
+            ProcessTb(MockConfig(mappings), test_logger)
```

Comment:
You should consider adding an `is_dir()` mock in this test:  we don't expect it to be invoked, so, if it does get called, it should fail the test.